flake.nix

```diff
@@ -42,10 +47,18 @@
               ])
               ++ [pkgs.z3];
 
-            ## Needed by the haskell-language-server
-            shellHook = ''
-              export LD_LIBRARY_PATH="${pkgs.z3.lib}/lib:''${LD_LIBRARY_PATH:+:}"
-            '';
+            inherit shellHook;
+          };
+
+          ## Lightweight development shell.
+          lwt = hpkgs.shellFor {
+            buildInputs = with pkgs; [ghc cabal-install z3];
+            packages = p: [
+              smtlib-backends
+              smtlib-backends-tests
+              smtlib-backends-z3
+            ];
```

Comment:
Maybe you can factor this list of packages, with something like 
```
let smtlib-packages = p:
  let smtlib-backends = p.callCabal2nix ... ;
       smtlib-backends-tests = p.callCabal2nix ...;
       smtlib-backends-z3 = p.callCabal2nix ...;
  in [ smtlib-backends smtlib-backends-tests smtlib-backends-z3 ]
```
This also ensures that if the `hpkgs` in `hpkgs.shellFor` is changed, the packages are rebuilt with the shell's version of GHC and not with the global `hpkgs = nixpkgs.haskellPackages`.